server/tasks/process_file.py

```diff
@@ -30,32 +34,82 @@ def process_file(serialized_agent: bytes):
     asyncio.run(agent.environment.finalize())
 
 
-@app.task(name="process_file_streaming", track_started=True, bind=True)
-def process_file_streaming(self, serialized_agent: bytes):
-    # Load the agent
-    agent = pickle.loads(serialized_agent)
+@app.task(
+    name="streaming_parent_task", track_started=True, bind=True, serializer="pickle"
+)
+def streaming_parent_task(
+    self, agent: Agent, result_handler: ResultHandler, batch_size: int = 2
+):
+    # Parent job ID is used for input/output topic names
+    parent_job_id = self.request.id
+
+    task = process_file_streaming
+    logger.info(f"Submitting task {task.name} with agent {agent}")
+    input_result = task.delay(agent=agent, parent_job_id=parent_job_id)
+    input_job_id = input_result.id
+    logger.info(f"Task {task.name} submitted with job_id {input_job_id}")
+
+    task = process_streaming_output
```

Comment:
Similar here, I think `relay_task` or `result_handler_task` would be more descriptive than `output_task`

Reply:
? its not called `output_task` but we can change it I guess

Reply:
Ooo the left hand side you mean, ok